### DISCUSSION TITLE
Nested types not supported by @DependencyClient

### DISCUSSION THREAD

**Discussion by malonehedges** [pointfreeco/swift-dependencies#136] [General]
### Description

```swift
@DependencyClient
struct SomeFailingClient {
  var endpoint: () async throws -> InternalValue
  
  struct InternalValue {
    var value: Bool
  }
}

extension SomeFailingClient.InternalValue {
}
```

<img width="1007" alt="image" src="https://github.com/pointfreeco/swift-dependencies/assets/12092296/81c945ad-d2cb-449c-84c6-dcd87f5ba20c">

### Checklist

- [X] I have determined whether this bug is also reproducible in a vanilla SwiftUI project.
- [ ] If possible, I've reproduced the issue using the `main` branch of this package.
- [X] This issue hasn't been addressed in an [existing GitHub issue](https://github.com/pointfreeco/swift-dependencies/issues) or [discussion](https://github.com/pointfreeco/swift-dependencies/discussions).

### Dependencies version information

1.1.0

### Destination operating system

iOS 17

### Swift Compiler version information

```shell
swift-driver version: 1.87.1 Apple Swift version 5.9 (swiftlang-5.9.0.128.108 clang-1500.0.40.1)
Target: arm64-apple-macosx14.0
```

**Comment by mbrandonw** [comment]
Hi @malonehedges, this is a bug in the compiler unfortunately. We have it documented in TCA, but perhaps we need to document it in this repo too:

https://pointfreeco.github.io/swift-composable-architecture/main/documentation/composablearchitecture/reducer()#Circular-reference-errors

As this is not an issue with the library I am going to convert it to a discussion.

**Reply by stephencelis** [reply]
You can follow the Apple bug here: https://github.com/apple/swift/issues/66450
